app/src/main/java/com/techcourse/dao/UserDao.java

```diff
@@ -1,121 +1,75 @@
+import com.interface21.jdbc.core.RowMapper;
+import com.techcourse.config.DataSourceConfig;
+import com.techcourse.domain.User;
+import java.util.List;
+import javax.sql.DataSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.sql.DataSource;
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.util.List;
-
 public class UserDao {
 
     private static final Logger log = LoggerFactory.getLogger(UserDao.class);
+    private static final RowMapper<User> rowMapper = (resultSet) -> new User(
+            resultSet.getLong(1),
+            resultSet.getString(2),
+            resultSet.getString(3),
+            resultSet.getString(4)
+    );
 
     private final DataSource dataSource;
+    private final JdbcTemplate jdbcTemplate;
 
     public UserDao(final DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
         this.dataSource = dataSource;
     }
 
     public UserDao(final JdbcTemplate jdbcTemplate) {
-        this.dataSource = null;
+        this.dataSource = DataSourceConfig.getInstance();
+        this.jdbcTemplate = new JdbcTemplate(this.dataSource);
     }
 
     public void insert(final User user) {
-        final var sql = "insert into users (account, password, email) values (?, ?, ?)";
-
-        Connection conn = null;
-        PreparedStatement pstmt = null;
-        try {
-            conn = dataSource.getConnection();
-            pstmt = conn.prepareStatement(sql);
-
-            log.debug("query : {}", sql);
-
-            pstmt.setString(1, user.getAccount());
-            pstmt.setString(2, user.getPassword());
-            pstmt.setString(3, user.getEmail());
-            pstmt.executeUpdate();
-        } catch (SQLException e) {
-            log.error(e.getMessage(), e);
-            throw new RuntimeException(e);
-        } finally {
-            try {
-                if (pstmt != null) {
-                    pstmt.close();
-                }
-            } catch (SQLException ignored) {}
-
-            try {
-                if (conn != null) {
-                    conn.close();
-                }
-            } catch (SQLException ignored) {}
-        }
+        String sql = String.format(
+                "insert into users (account, password, email) values ('%s', '%s', '%s')",
+                user.getAccount(), user.getPassword(), user.getEmail());
+
+        jdbcTemplate.update(sql);
     }
 
     public void update(final User user) {
-        // todo
+        var sql = String.format("""
+                update 
+                    users 
+                set
+                    account = '%s',
+                    password = '%s', 
+                    email = '%s'
+                where 
+                    id = %d
+                """, user.getAccount(), user.getPassword(), user.getEmail(), user.getId());
+
+        jdbcTemplate.update(sql);
```

Comment:
**첫번째**로 SQL 구성에 대해서는 사용자에게 책임이 있다고 생각합니다. JdbcTemplate을 쓰더라도 **SQL구성하는 책임은 사용자에게 있는 것은 다름 없습니다.** (`select * from users from account =` 까지는 사용자가 구성해야되는 부분이니까요. ) 

**두번째**로 `안전한 쿼리구성`에 대한 이야기입니다.
제가 생각하는 JdbcTemplate은 반복되는 작업을 최소화 하기위한 라이브러리 정도입니다. 물론 저도 ?를 활용해서 인자를 받아서 sql을 완벽하게 구성하는 것을 JdbcTemplate에 넘기도록 할까 고민은 했지만, 그러면 SQL 구성하는 책임은 사용자인지? 라이브러리인지? 모호하다고 생각해서 JdbcTemplate의 스펙을 따르지 않고 직접 그 이유를 느껴보기로 했습니다.

콜리의 질문을 받고 JdbcTemplate의 공식문서를 봤습니다. (물론 미션시작할 때 참조되어 그때도 봤지만..그냥 넘어갔던,,)
```This class simplifies the use of JDBC and helps to avoid common errors.```

제가 직접 구현할 때도 느꼈지만, 직접 SQL 구성하는 것은 휴먼에러가 자주 생깁니다. 작은 따옴표를 빼먹는 실수가 있었죠.
그러한 면에서 JdbcTemplate은 사용자가 자주 발생시키는 SQL 문법오류를 줄여주는데 도움을 주는 것이 맞는 것 같습니다. 하지만 모든 where절을 제외한 오류는 잡아주지 않는다는 점에서 **'안전한 쿼리구성' 은 jdbcTemplate을 써도 크게 달라지지 않는다고 생각합니다.** (저는 그러면 안되지만 아직 update set 쿼리를 자주 틀리거든요,,)

**세번째**로 **`SqlInjection`은 확실하게 PreparedStatement가 방지는 해줍니다.** 좋은 포스트 추천감사합니다. 이러한 면에서 JdbcTemplate이 지원해주는 것은 좋은 것 같네요.

**마지막**으로 하고싶은 얘기는 `하지만, 내가 구성한 메서드도 JdbcTemplate의 스펙이다` 입니다. 저는 SqlInjection과 Where절의 문법 오류를 방지하지 못한 JdbcTemplate을 만들었을 뿐 **저의 구현도 JdbcTemplate의 스펙에 존재한다는 점입니다.**

이점은 PreparedStatement를 사용하는 것이 더 많겠네요! 이것도 한번 구현해볼게요~!